go-app.js

```diff
@@ -35,11 +41,39 @@ go.app = function() {
             return new FreeText(name, {
                 question: 'Please enter a location on National, Provincial or Ward level to query:',
                 next: function(content) {
-                    return 'states:end';
+                    return self 
+                    .http.get('http://wazimap.co.za/place-search/json/', {
+                        params: {q : content}
+                    })
+                    .then (function(response) {  
+                        return {
+                            name: 'states:results',
+                            creator_opts: {
+                                locations: response.data.results
+                            } 
+                        };
+                    }); 
                 }   
             });
         });
-              
+
+
+        self.states.add('states:results', function(name, opts) {
+			var location_choices = _.map(opts.locations.slice(0, 2), function(d) {
+                return new Choice(d.full_geoid, d.full_name);
+			});
+
+			return new ChoiceState(name, {
+				question: 'Please select the location you would like to query:',
+				choices: location_choices,
+				characters_per_page: 160,
+				options_per_page : 3,
+				next: function(choice) {
+					return choice.value;
+					}	
+				});
+			});
```

Comment:
A lot of this function seems over-indented. Could you indent it to the normal level (it might be `tab`s that are causing the indentation to look weird, so convert those to spaces).